tests/unit_tests/corsika/test_corsika_histograms.py

```diff
@@ -783,52 +784,28 @@ def test_dict_1D_distributions(corsika_histograms_instance_set_histograms):
-        "hist_1D_photon_radial_distr_all_tels.ecsv",
-        "hist_1D_photon_density_distr_all_tels.ecsv",
-        "hist_1D_photon_time_distr_all_tels.ecsv",
-        "hist_1D_photon_time_distr_all_tels.ecsv",
-        "hist_1D_photon_per_event_distr_all_tels.ecsv",
-        "hist_1D_photon_per_telescope_distr_all_tels.ecsv",
-    ]:
-        assert io_handler.get_output_directory(dir_type="test").joinpath(file_name).exists()
-
-
-def test_export_2D_histograms(corsika_histograms_instance_set_histograms, io_handler):
-    corsika_histograms_instance_set_histograms._export_2D_histograms()
-
-    for file_name in [
-        "hist_2D_photon_direction_distr_all_tels.ecsv",
-        "hist_2D_photon_time_altitude_distr_all_tels.ecsv",
-        "hist_2D_photon_telescope_event_distr_all_tels.ecsv",
-        "hist_2D_photon_count_distr_all_tels.ecsv",
-        "hist_2D_photon_density_distr_all_tels.ecsv",
-    ]:
-        assert io_handler.get_output_directory(dir_type="test").joinpath(file_name).exists()
+def test_export_and_read_histograms(corsika_histograms_instance_set_histograms, io_handler):
+    # Default values
+    corsika_histograms_instance_set_histograms.export_histograms()
 
+    file_name = Path(corsika_histograms_instance_set_histograms.output_path).joinpath(
+        "tel_output_10GeV-2-gamma-20deg-CTAO-South.hdf5"
+    )
+    assert io_handler.get_output_directory(dir_type="test").joinpath(file_name).exists()
 
-def test_export_histograms(corsika_histograms_instance_set_histograms, io_handler):
+    # Change hdf5 file name
+    corsika_histograms_instance_set_histograms.hdf5_file_name = "test.hdf5"
     corsika_histograms_instance_set_histograms.export_histograms()
+    output_file = io_handler.get_output_directory(dir_type="test").joinpath("test.hdf5")
+    assert output_file.exists()
 
-    for file_name in [
-        "hist_1D_photon_wavelength_distr_all_tels.ecsv",
-        "hist_1D_photon_radial_distr_all_tels.ecsv",
-        "hist_1D_photon_density_distr_all_tels.ecsv",
-        "hist_1D_photon_time_distr_all_tels.ecsv",
-        "hist_1D_photon_time_distr_all_tels.ecsv",
-        "hist_1D_photon_per_event_distr_all_tels.ecsv",
-        "hist_1D_photon_per_telescope_distr_all_tels.ecsv",
-        "hist_2D_photon_direction_distr_all_tels.ecsv",
-        "hist_2D_photon_time_altitude_distr_all_tels.ecsv",
-        "hist_2D_photon_telescope_event_distr_all_tels.ecsv",
-        "hist_2D_photon_count_distr_all_tels.ecsv",
-        "hist_2D_photon_density_distr_all_tels.ecsv",
-    ]:
-        assert io_handler.get_output_directory(dir_type="test").joinpath(file_name).exists()
+    # Read hdf5 file
+    list_of_tables = corsika_histograms_instance_set_histograms.read_hdf5(output_file)
+    assert len(list_of_tables) == 12
+    for table in list_of_tables:
+        assert isinstance(table, Table)
+    # Check piece of metadata
+    assert list_of_tables[-1].meta["corsika_version"] == 7.741
```

Comment:
Could you use here the value from the metadata before writing (earlier in the function) instead of putting the corsika version to the last digit into the file?

Reply:
yes, I changed it now. now we dont have to change this test everytime the corsika version changes.